makeflow/src/makeflow_status.c

```diff
@@ -0,0 +1,268 @@
+	x = jx_lookup_string(*a, "project");
+	if(!x)
+		x = "unknown";
+
+	y = jx_lookup_string(*b, "project");
+	if(!y)
+		y = "unknown";
+
+	result = strcasecmp(x, y);
+	if(result != 0)
+		return result;
+        
+        //by owner
+        x = jx_lookup_string(*a, "owner");
+	if(!x)
+		x = "unknown";
+
+	y = jx_lookup_string(*b, "owner");
+	if(!y)
+		y = "unknown";
+
+	result = strcasecmp(x, y);
+	if(result != 0)
+		return result;
+
+        //failsafe by servername
+	x = jx_lookup_string(*a, "name");
+	if(!x)
+		x = "unknown";
+
+	y = jx_lookup_string(*b, "name");
+	if(!y)
+		y = "unknown";
+
+	return strcasecmp(x, y);
+}
+
+static struct jx *table[10000];
+
+int first_comma(char* in){
+    int i=0;
+    int max = strlen(in);
+    for(i=0; i<max; ++i){
+        if(in[i] == ',')
+            return i;
+    }
+    return -1;
+}
+char* add_port(char* in, int port){
+    char buffer[256];
+    int start = 0;
+    char* cpy = string_format("%s",in);
+    int com = first_comma(cpy);
+    char* retstr=NULL;
+    
+    //in the event there is no comma, and it's a single address
+    if(com == -1){
+        return string_format("%s:%i",in,port);
+    }
+    //in the event that there is a comma
+    while(com > 0){
+        //reset buffer
+        memset(buffer,0,256);
+        strncpy(buffer,in+start,com-start);
+        char* tmp;
+        if(retstr != NULL){
+            tmp=string_format("%s,%s:%i",retstr,buffer,port);
+            free(retstr);
+        }else{
+            tmp=string_format("%s:%i",buffer,port);
+        }
+        retstr = tmp;
+        
+        cpy[com]=' ';
+        start = com+1;
+        com = first_comma(cpy);
+    }
+    free(cpy);
+    return retstr;
+    
+}
+
+/*
+ * Obtains information from the Catalog, format it, and make return it to user.
+ */
+int main(int argc, char** argv) {
+
+    static const struct option long_options[] = {
+        {"help", no_argument, 0, 'h'},
+        {"port", required_argument, 0, 'p'},
+        {"project", required_argument, 0, 'N'},
+        {"server", required_argument, 0, 's'},
+        {"timeout", required_argument, 0, 't'},
+        {"username", required_argument, 0, 'u'},
+        {0, 0, 0, 0}
+    };
+
+    struct catalog_query *q;
+    struct jx *j;
+    int c;
+    unsigned int i;
+    time_t timeout = 60;
+    char* catalog_host = NULL;
+    
+    char* username = NULL;
+    char* project = NULL;
+    
+    char* server = NULL;
+    int port = 0;
+    long long porthelp = 0;
+    
+    while ((c = getopt_long(argc, argv, "N:t:u:w:p:s:h", long_options, NULL)) > -1) {
+        switch (c) {
+            case 'N':
+                project = xxstrdup(optarg);
+                break;
+            case 't':
+                timeout = string_time_parse(optarg);
+                break;
+            case 'u':
+                username = xxstrdup(optarg);
+                break;
+            case 's':
+                server = xxstrdup(optarg);
+                break;
+            case 'p':
```

Comment:
I don't think there should be a `-p` option as it doesn't play nicely with multiple catalog servers. The `-s` option could hold a host list, then you don't have to worry about ports. The user can choose any host/port combos, e.g.
- `"server1"`
- `"server1:2000"`
- `"server1,server2:2232,server3"`